Source/WebCore/loader/MixedContentChecker.cpp

```diff
@@ -53,53 +53,44 @@ bool MixedContentChecker::isMixedContent(SecurityOrigin& securityOrigin, const U
 {
-    const char* errorString = allowed ? " was allowed to " : " was not allowed to ";
-    auto message = makeString((allowed ? "" : "[blocked] "), "The page at ", frame.document()->url().stringCenterEllipsizedToLength(), errorString, action, " insecure content from ", target.stringCenterEllipsizedToLength(), ".\n");
+    const char* errorString = !blocked ? "automatically upgraded and should" : "blocked and must";
+    auto message = makeString((!blocked ? "" : "[blocked] "), "The page at ", frame.document()->url().stringCenterEllipsizedToLength(), " requested insecure content from ", target.stringCenterEllipsizedToLength(), ". This content was ", errorString, " be served over HTTPS.\n");
     frame.document()->addConsoleMessage(MessageSource::Security, MessageLevel::Warning, message);
 }
 
-bool MixedContentChecker::canDisplayInsecureContent(Frame& frame, SecurityOrigin& securityOrigin, ContentType type, const URL& url, AlwaysDisplayInNonStrictMode alwaysDisplayInNonStrictMode)
+bool MixedContentChecker::shouldUpgradeInsecureContent(SecurityOrigin& securityOrigin, ContentType type, const URL& url, FetchOptions::Mode mode, Initiator initiator)
 {
-    if (!isMixedContent(securityOrigin, url))
-        return true;
-
-    if (!frame.document()->contentSecurityPolicy()->allowRunningOrDisplayingInsecureContent(url))
+    if (!isMixedContent(securityOrigin, url) || type != ContentType::Upgradable)
         return false;
 
-    bool isStrictMode = frame.document()->isStrictMixedContentMode();
-    if (!isStrictMode && alwaysDisplayInNonStrictMode == AlwaysDisplayInNonStrictMode::Yes)
-        return true;
-
-    bool allowed = !isStrictMode && (frame.settings().allowDisplayOfInsecureContent() || type == ContentType::ActiveCanWarn) && !frame.document()->geolocationAccessed();
-    logWarning(frame, allowed, "display"_s, url);
-
-    if (allowed) {
-        frame.document()->setFoundMixedContent(SecurityContext::MixedContentType::Inactive);
-        frame.loader().client().didDisplayInsecureContent();
-    }
-
-    return allowed;
+    return url.protocolIs("http"_s)
+        && initiator != Initiator::Imageset
+        && mode != FetchOptions::Mode::Cors
+        && !URL::hostIsIPAddress(url.host());
 }
 
-bool MixedContentChecker::canRunInsecureContent(Frame& frame, SecurityOrigin& securityOrigin, const URL& url)
+bool MixedContentChecker::shouldBlockInsecureContent(Frame& frame, SecurityOrigin& securityOrigin, ContentType type, const URL& url, FetchOptions::Mode mode, Initiator initiator)
 {
     if (!isMixedContent(securityOrigin, url))
-        return true;
-
-    if (!frame.document()->contentSecurityPolicy()->allowRunningOrDisplayingInsecureContent(url))
         return false;
 
-    bool allowed = !frame.document()->isStrictMixedContentMode() && frame.settings().allowRunningOfInsecureContent() && !frame.document()->geolocationAccessed() && !frame.document()->secureCookiesAccessed();
```

Comment:
Are we sure we do not change WK1 behavior here?
Maybe we should split the removal of allowRunningOfInsecureContent and didRunInsecureContent call in its own patch?

Reply:
Okay, I'll do that. I found we only used this WK1 setting on WK1 Windows, but we can preserve it here and clean up in another patch for safety.

Reply:
@youennf I think it's safe to remove allowRunningOfInsecureContent and didRunInsecureContent immediately. They no longer have any value after this merge request. The only reason we would need to keep them is if we wind up with some regression and have to revert this whole commit anyway.

Reply:
Anyway if we remove them at the same time, then https://bugs.webkit.org/show_bug.cgi?id=219396 can be a duplicate of https://bugs.webkit.org/show_bug.cgi?id=247197. Otherwise, I can use 219396 to do this follow-up.